### PR TITLE
<DatePicker/> - Add ability to use external locales in [WIP]

### PR DESCRIPTION
**Pull request by nickg-wix**
### What changed
Added new DatePicker prop - `locales`. By using it customers could pass additional translations from [date-fns](https://github.com/date-fns/date-fns) library to DatePicker.
...

### Why it changed
Issue: [#1679](https://github.com/wix/wix-style-react/issues/1679)

---

- [x] Change is tested
- [ ] Change is documented
- [ ] Build is green

### CODE REVIEW

File: src/DatePicker/DatePicker.js [hunk 1]
```diff
@@ -53,23 +53,10 @@ export default class DatePicker extends WixComponent {
...
-      'ru',
-      'ja',
-      'ko',
-      'tr',
-      'sv',
-      'no',
-      'nl',
-      'da'
-    ]),
+    locale: PropTypes.string,
```
Comment (nickg-wix) [inline src/DatePicker/DatePicker.js]: Don't really like this change, but seems thats the price we need to pay if want custom translations

Comment (argshook) [reply src/DatePicker/DatePicker.js]: @nickg-wix what if we do not add `locales` but instead extend `locale` prop to be `PropTypes.oneOf(['en', 'es' ..., PropTypes.shape(/* shape of locale object */)])`.

this way API remains the same but consumer is able to pass custom locale
if think consumer doesn't need to provide multiple locales at once. if they need to change it dynamically, they can pass different locale to `locale`

wdyt? would it be possible to do without changing too much of what you already did?

Comment (nickg-wix) [reply src/DatePicker/DatePicker.js]: Nice suggestion! Will change to this approach

File: src/DatePicker/DatePicker.js [hunk 2]
```diff
@@ -53,22 +54,27 @@ export default class DatePicker extends WixComponent {
...
-      'ru',
-      'ja',
-      'ko',
-      'tr',
-      'sv',
-      'no',
-      'nl',
-      'da'
+    locale: PropTypes.oneOfType([
+      PropTypes.oneOf([
```
Comment (nickg-wix) [inline src/DatePicker/DatePicker.js]: `oneOf` seems to be limited to simple literal enum, it doesn't support mixed values.

**Comment by nickg-wix** [comment]
On hold because of idea to use `Intl` API.

**Comment by Varnius** [comment]
Investigated Intl API with a conclusion that it's not really an option atm because browser implementation of Intl is inconsistent with Node Intl polyfill (eg. produces different month names compared to browser under certain locales). This can lead to problems with SSR and tests run on node (jsdom). Also, Intl options of formatting dates are much more limited compared to date-fsn.
We should also update the version of date-fns we use, after they release a stable date-fsn v2.

**Comment by shlomitc** [comment]
@argshook - what should we do with this PR?

**Comment by argshook** [comment]
@shlomitc i have small tweaks and will merge

**Pull request merged**